src/wp-includes/rest-api.php

```diff
@@ -1776,3 +1772,132 @@ function rest_filter_response_by_context( $data, $schema, $context ) {
 
 	return $data;
 }
+
+
+/**
+ * Outputs a header link to a REST resource
+ *
+ * @since 5.5.0
+ *
+ * @param string       $url URL.
+ */
+function rest_head_link( $url ) {
+	if ( ! empty( $url ) ) {
+		printf( '<link rel="alternate" type="application/json" href="%s" />', esc_url( $url ) );
+	}
+}
+
+/**
+ * Return the rest route for a post.
+ *
+ * @since 5.5.0
+ *
+ * @param WP_Post       $post Post Object.
+ * @return string           Route.
+ */
+function rest_get_route_for_post( $post ) {
+	if ( ! $post instanceof WP_Post ) {
+		return '';
+	}
+
+	$post_type = get_post_type_object( $post->post_type );
+	if ( ! $post_type ) {
+		return '';
+	}
+
+	$route = '';
+	$controller = $post_type->get_rest_controller();
+	if ( ! $controller ) {
+		return '';
+	}
+        
+	// The only two controllers that we can detect are the Attachments and Posts controllers.
+	if ( in_array( get_class( $controller ), array( 'WP_REST_Attachments_Controller', 'WP_REST_Posts_Controller' ), true ) ) {
+		$namespace = 'wp/v2';
+		$rest_base = ! empty( $post_type->rest_base ) ? $post_type->rest_base : $post_type->name;
+		$route = sprintf( '%s/%s/%d', $namespace, $rest_base, $post->ID );
+	}
+
+	/**
+	 * Filters the route based on information in the post object.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $route Route.
+	 * @param WP_Post $post Post Object.
+	 */
+	return apply_filters( 'rest_route_for_post', $route, $post );
+}
+
+/**
+ * Return the rest route for a term.
+ *
+ * @since 5.5.0
+ *
+ * @param WP_Term       $term Term.
+ * @return string           Route.
+ */
+function rest_get_route_for_term( $term ) {
+	if ( ! $term instanceof WP_Term ) {
+		return '';
+	}
+	
+	$taxonomy = get_taxonomy( $term->taxonomy );
+	if ( ! $taxonomy ) {
+		return '';
+	}
+
+	$route = '';
+	$controller = $taxonomy->get_rest_controller();
+	if ( ! $controller ) {
+		return '';
+	}
+
+	// The only controller that works is the Terms controller.
+	if ( 'WP_REST_Terms_Controller' === get_class( $controller ) ) {
+		$namespace = 'wp/v2';
+		$rest_base = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxnomomy->name;
+		$route = sprintf( '%s/%s/%d', $namespace, $rest_base, $term->term_id );
+	}
+
+	/**
+	 * Filters the route based on information in the term object.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $route Route.
+	 * @param WP_Term $term Term Object.
+	 */
+	return apply_filters( 'rest_route_for_term', $route, $term );
+}
+
+/**
+ * Get a REST URL for the current object.
+ *
+ * @param string URL.
+ */
+function rest_get_current_resource_link() {
+	if ( is_singular() ) {
+		return rest_url( rest_get_route_for_post( get_post() ) );
+	}
+
+	if ( is_category() || is_tag() || is_tax() ) {
+		return rest_url( rest_get_route_for_term( get_queried_object() );
```

Comment:
```suggestion
		return rest_url( rest_get_route_for_term( get_queried_object() ) );
```

I believe this is what is causing the tests to fatal error, https://travis-ci.com/github/WordPress/wordpress-develop/jobs/348826497#L1196